train.lua

```diff
@@ -32,349 +35,195 @@ cmd:text('Train a character-level language model')
 cmd:text()
 cmd:text('Options')
 -- data
-cmd:option('-data','tinyshakespeare','name of data directory. Should contain the file input.txt with input data')
-cmd:option('-back','cuda','cpu|cuda|cl')
-cmd:option('-len', -1, 'only use this many characters from input data. -1 for all data')
+cmd:option('-data', 'tinyshakespeare', 'name of data directory. Should contain the file input.txt with input data')
+cmd:option('-back', 'cuda', 'cpu|cuda|cl')
 cmd:option('-seq', 50, 'sequence length to use')
-cmd:option('-dump', 100, 'iterations between writing out weights')
 cmd:option('-hidden', '128,128', 'size of hidden layers, comma-separated, one per required hidden layer')
-cmd:option('-drop',0 , 'dropout probability')
-cmd:option('-batch',1 , 'batch size (should default to 50 probably)')
```

Comment:
I hate typing really long commandlines, therefore made the options relatively succinct.  'batch' instead of 'batchsize'.  It's a bit subjective, but that was kind of my reasonining...